src/axom/spin/internal/linear_bvh/build_radix_tree.hpp

```diff
@@ -364,15 +382,20 @@ void reorder(int32* indices, T*&array, int32 size)
 template < typename ExecSpace >
 void custom_sort( ExecSpace, uint32*& mcodes, int32 size, int32* iter )
 {
+  AXOM_PERF_MARK_FUNCTION( "custom_sort" );
+
   array_counting< ExecSpace >(iter, size, 0, 1);
 
-  std::stable_sort( iter,
-                    iter + size,
-                    [=](int32 i1, int32 i2)
-                    {
-                    return mcodes[i1] < mcodes[i2];
-                    } );
+  AXOM_PERF_MARK_SECTION( "cpu_sort",
```

Comment:
It seems a bit intrusive and limiting to me for the ``SECTION`` annotation/mark to wrap the call to a function, as opposed to having an explicit begin/end call.  E.g. can this annotation support more than a single function call?

I suppose there is a tradeoff and we can support both types once ``caliper`` gets in.

Reply:
Yes -- it's a variadic macro, you can pass whatever block of code.

I think this is much cleaner than littering the code with begin/end annotations.

The only current limitation is that you cannot "easily" nest sections. I don't think this would be common, but it can be fixed by another macro if/when it becomes important.